src/rdkafka_admin.c

```diff
@@ -4108,6 +4109,291 @@ rd_kafka_ListOffsetsResponse_parse(rd_kafka_op_t *rko_req,
+                rd_kafka_buf_read_i16(reply, &top_level_error_code);
+        }
+
+        /* #partitions */
+        rd_kafka_buf_read_arraycnt(reply, &TopicArrayCnt, RD_KAFKAP_TOPICS_MAX);
+
+        rd_list_init(&partitions_arr, TopicArrayCnt,
+                     rd_kafka_topic_partition_result_free);
+
+        for (i = 0; i < TopicArrayCnt; i++) {
+                rd_kafka_topic_partition_result_t *partition_result;
+                rd_kafkap_str_t ktopic;
+                char *topic                  = NULL;
+                int32_t partition            = -1;
+                int16_t partition_error_code = 0;
+                rd_kafkap_str_t partition_error_msg;
+                char *partition_errstr;
+                int32_t PartArrayCnt;
+
+                rd_kafka_buf_read_str(reply, &ktopic);
+                RD_KAFKAP_STR_DUPA(&topic, &ktopic);
+
+                rd_kafka_buf_read_arraycnt(reply, &PartArrayCnt,
+                                           RD_KAFKAP_PARTITIONS_MAX);
+
+                while (PartArrayCnt-- > 0) {
+                        rd_kafka_buf_read_i32(reply, &partition);
+                        rd_kafka_buf_read_i16(reply, &partition_error_code);
+                        rd_kafka_buf_read_str(reply, &partition_error_msg);
+
+                        rd_kafka_buf_skip_tags(reply);
+
+                        if (partition_error_code) {
+                                if (RD_KAFKAP_STR_IS_NULL(
+                                        &partition_error_msg) ||
+                                    RD_KAFKAP_STR_LEN(&partition_error_msg) ==
+                                        0)
+                                        partition_errstr =
+                                            (char *)rd_kafka_err2str(
+                                                partition_error_code);
+                                else
+                                        RD_KAFKAP_STR_DUPA(
+                                            &partition_errstr,
+                                            &partition_error_msg);
+                        }
+                        partition_result = rd_kafka_topic_partition_result_new(
+                            topic, partition, partition_error_code,
+                            partition_errstr);
+
+                        rd_list_add(&partitions_arr, partition_result);
+                }
+                rd_kafka_buf_skip_tags(reply);
+        }
+
+        rd_kafka_buf_skip_tags(reply);
+
+        result = rd_kafka_ElectLeadersResult_new(top_level_error_code,
+                                                 &partitions_arr);
+
+        rko_result = rd_kafka_admin_result_new(rko_req);
+
+        rd_list_init(&rko_result->rko_u.admin_result.results, 1,
+                     rd_kafka_ElectLeadersResult_free);
+
+        rd_list_add(&rko_result->rko_u.admin_result.results, result);
+
+        *rko_resultp = rko_result;
+
+        rd_list_destroy(&partitions_arr);
```

Comment:
`partition_arr` should also be freed in `err_parse` case. `partition_arr` should be initialized with `NULL`. Use index to determine how many to free.